drivers/input/input_gpio_keys.c

```diff
@@ -123,6 +130,13 @@ static int gpio_keys_init(const struct device *dev)
 	const struct gpio_keys_config *cfg = dev->config;
 	int ret;
 
+#ifdef CONFIG_PINCTRL
+	ret = pinctrl_apply_state(cfg->pinctrl_cfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		return ret;
+	}
```

Comment:
this will make the driver useless to any platform that has CONFIG_PINCTRL=y and no pinctrl entries for each gpio-keys subnode.

Reply:
What is your suggestion about how to handle it?

Reply:
see my comment above